### PR TITLE
MM-30712 - Restart update job if image changed

### PR DESCRIPTION
**Pull request by Szymongib**
#### Summary

When updating Cluster Installation with an invalid image and the update job fails it needs to be cleaned up manually.
To remove a need for manual deletion, the Operator will now handle the scenario when the desired image changed.
Changes in the PR:
- Check if the job image matches the desired one
- Restart job if image changed

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-30712

#### Release Note

```release-note
Restart update job if image name changed
```

### CODE REVIEW

**Review by jwilander — APPROVED** [review]
LGTM 👍

File: controllers/mattermost/clusterinstallation/mattermost.go [hunk 1]
```diff
@@ -310,6 +310,27 @@ func (r *ClusterInstallationReconciler) launchUpdateJob(
...
 
+// restartUpdateJob removes existing update job if it exists and creates new one.
+func (r *ClusterInstallationReconciler) restartUpdateJob(
+	mi *mattermostv1alpha1.ClusterInstallation,
+	currentJob *batchv1.Job,
+	deployment *appsv1.Deployment,
+) error {
+	err := r.Client.Delete(context.TODO(), currentJob, k8sClient.PropagationPolicy(metav1.DeletePropagationBackground))
+	if err != nil && !k8sErrors.IsNotFound(err) {
+		return errors.Wrapf(err, "failed to delete outdated update job")
```
Comment (gabrieljackson) [inline controllers/mattermost/clusterinstallation/mattermost.go:321]:
```suggestion
		return errors.Wrap(err, "failed to delete outdated update job")
```

**Review by gabrieljackson — APPROVED** [review]
Looks good!

**Comment by jwilander** [comment]
Merging after the tests pass so it gets included in the next release

**Pull request merged**